html/admin/api/server/migrateAsset.php

```diff
@@ -0,0 +1,101 @@
+<?php
+require_once __DIR__ . '/../apiHeadSecure.php';
+
+if (!$AUTH->serverPermissionCheck("INSTANCES:FULL_PERMISSIONS_IN_INSTANCE") or !isset($_POST['assets_id']) or !isset($_POST['instances_id'])) die("404");
+
+//Get asset
+$DBLIB->where("assets_id", $_POST['assets_id']);
+$DBLIB->where("assets_deleted", 0);
+$DBLIB->where("assets_archived IS NULL");
+$asset = $DBLIB->getOne("assets");
+if (!$asset) finish(false, ["code"=>"ASSET_NOT_FOUND", "message"=>"Asset not found"]);
+
+//Get assetType
+$DBLIB->where("assetTypes_id", $asset['assetTypes_id']);
+$assetType = $DBLIB->getOne("assetTypes", ["assetTypes.*"]);
+if (!$assetType) finish(false, ["code"=>"ASSET_TYPE_NOT_FOUND", "message"=>"Asset type not found"]);
+
+//Check if asset is already in instance
+if ($assetType['instances_id'] == $_POST['instances_id']) finish(false, ["code"=>"ASSET_ALREADY_IN_INSTANCE", "message"=>"Asset already in instance"]);
+
+if ($assetType['instances_id'] !== NULL) {
+    //See if this instance has an asset with a similar name
+    $DBLIB->where("assetTypes_name", "%" . $assetType['assetTypes_name'] . "%", 'LIKE');
+    $DBLIB->where("instances_id", $_POST['instances_id']);
+    $possibleAssetType = $DBLIB->getOne("assetTypes", ["assetTypes.*"]);
+    if ($possibleAssetType) {
+        $assetType = $possibleAssetType;
```

Comment:
I'm not sure if this is desirable behaviour - it would be fine I think if you had an exact match on the asset type name, but otherwise this is a very loose match and they could be quite different asset types (ie "Source Four LED" would be copied across as a "LED worklight")